enclave_manager/avalon_enclave_manager/kme_listener.py

```diff
@@ -0,0 +1,290 @@
+#!/usr/bin/env python3
+
+# Copyright 2020 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import sys
+import json
+import logging
+import argparse
+
+from urllib.parse import urlsplit
+from twisted.web import server, resource, http
+from twisted.internet import reactor, error as reactor_error
+from jsonrpc import JSONRPCResponseManager
+from jsonrpc.dispatcher import Dispatcher
+
+import config.config as pconfig
+import utility.logger as plogger
+from error_code.error_status \
+    import WPERegisterStatus, WOPreprocessStatus, JRPCErrorCodes
+from utility.jrpc_utility import create_error_response
+
+logger = logging.getLogger(__name__)
+
+TCFHOME = os.environ.get("TCF_HOME", "../../")
+
+
+class KMEListener(resource.Resource):
```

Comment:
To avoid duplicate code, please create a BaseListener class in common/python package and let Avalon Listener and KME Listener inherit from BaseListener